File: javascript/armstrong-numbers/armstrong-numbers.js
//
// This is only a SKELETON file for the 'Armstrong Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isArmstrongNumber = (num) => {
  
  let s; 
  const str = num.toString();
  let amountOfDigits;
  let convert;

  if (typeof(num) === 'bigint') {
    amountOfDigits = BigInt(str.length);
    s = 0n;
    convert = BigInt;
  } else {
    amountOfDigits = str.length
    s = 0;
    convert = parseInt;
  }

  for (let i = 0; i < amountOfDigits; i++) {
    let d = convert(str[i]);
    s += d ** amountOfDigits;
  }
  return s == num;

};

File: lua/armstrong-numbers/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  if number == 0 then
	  return true
  end

  local str = tostring(number)
  local amount_of_digits = #str
  local s = 0
  for dstr in string.gmatch(str, '.') do
    local d = tonumber(dstr) ^ amount_of_digits
    s = s + d
  end
  return s == number
end

return ArmstrongNumbers
